lib/use-cases/check-auth.js

```diff
@@ -0,0 +1,12 @@
+class CheckAuth {
+  constructor({allowedGroups, jwt}) {
+    this.allowedGroups = allowedGroups;
+    this.jwt = jwt;
+  }
+
+  execute({token}) {
+    const payload = this.jwt.decode(token)
+    return Boolean(payload) && Boolean(payload.groups) && payload.groups.some(g => this.allowedGroups.includes(g));
```

Comment:
I think `decode` will throw if it can't decode the token - rather than returning `null`? So you could remove your first null check and simplify this to:

```suggestion
    return Boolean(payload.groups) && payload.groups.some(g => this.allowedGroups.includes(g));
```

Personal preference mostly, but `payload.groups` but you don't need to explicitly turn `payload.groups` into a bool either - any `undefined` or `null` value will be falsy in this statement. So you could simplify this even more to something like:

```suggestion
    return payload.groups && payload.groups.some(g => this.allowedGroups.includes(g));
```

Reply:
Also, probably overkill but if `allowedGroups` is likely to contain fewer items that `payload.groups` you could swap the checks around to reduce the number of times this has to loop.

```suggestion
    return payload.groups && this.allowedGroups.some(g => payload.groups.includes(g));
```

Reply:
I tried without Boolean but it didn't like that